### PR TITLE
WIP: partial_fit in search light

### PR DESCRIPTION
**Pull request by kingjr**
This will allow fitting on subsets of the data to avoid having to load all epochs in RAM to do the fitting.

### CODE REVIEW

**Comment by kingjr** [comment]
I won't implement this for CSP and Xdawn, see #3483.

**Comment by agramfort** [comment]
can you give a usage example. I don't see the big picture and how it helps you.

**Comment by kingjr** [comment]
RAM is currently a bottleneck when fitting multiple runs of data.

This trick will allow us to scale up:
```
clf = SearchLight(SGDClassifier())
for block in range(10): 
     raw = Raw(fname % block)
     events = find_events(raw)
     epochs = epochs(raw, events)
     X = time_frequency_transform(epochs._data, epochs.info['sfreq'], np.logspace(1, 2))
     y = epochs.events[:, 2]
     clf.partial_fit(X, y)
```

File: mne/decoding/transformer.py [hunk 1]
```diff
@@ -672,15 +673,38 @@ def fit(self, X, y=None):
...
+        y : None | array, shape (n_samples,)
+            Used for scikit-learn compatibility.
+
+        Returns
+        -------
+        self : Instance of UnsupervisedSpatialFilter
+            Return the modified instance.
+        """
+        _check_partial_fit(self.estimator)
+        self.estimator.fit(self._preproc_X(X))
```
Comment (agramfort) [inline mne/decoding/transformer.py]: why not  self.estimator.partial_fit ?

Comment (kingjr) [inline mne/decoding/transformer.py]: error on my part

**Comment by kingjr** [comment]
@agramfort ready for review

**Comment by kingjr** [comment]
[wait for next release before merge]

**Comment by agramfort** [comment]
then call the PR WIP :)

**Comment by jona-sassenhagen** [comment]
Is this ready now?

**Comment by kingjr** [comment]
SearchLight has been privatized, so it'd need to be rebase, but else yes; I'm working on something else, do you need it?

**Comment by agramfort** [comment]
@kingjr can we close this?

**Comment by jona-sassenhagen** [comment]
(I don't *need* it, was just interested in speeding things up.)

**Comment by kingjr** [comment]
I think it needs testing to check the usecase before we commit

**Pull request closed without merging**